sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/FlatteningSerializer.java

```diff
@@ -158,7 +226,53 @@ public void serialize(Object value, JsonGenerator jgen, SerializerProvider provi
             jgen.writeNull();
             return;
         }
+
+        // Investigation into serializing the class ourselves to prevent string manipulation.
+        // Code below doesn't handle the following:
+        // - Null values are serialized without respecting ObjectMapper inclusion.
+        // - JsonType information isn't included.
+//        if (defaultSerializer instanceof BeanSerializer) {
+//            BeanSerializer beanSerializer = (BeanSerializer) defaultSerializer;
+//
+//            jgen.writeStartObject();
+//
+//            Iterator<PropertyWriter> writerIterator = beanSerializer.properties();
+//            while (writerIterator.hasNext()) {
+//                BeanPropertyWriter writer = (BeanPropertyWriter) writerIterator.next();
+//                String jsonPropertyName = writer.getName();
+//                if ((writer.getMember().hasAnnotation(JsonFlatten.class) || classHasJsonFlatten)
+//                    && CHECK_IF_FLATTEN_PROPERTY_PATTERN.matcher(jsonPropertyName).matches()) {
+//                    String[] flatteningNames = SPLIT_FLATTEN_PROPERTY_PATTERN.split(jsonPropertyName);
+//
+//                    jgen.writeFieldName(flatteningNames[0]);
+//                    jgen.writeStartObject();
+//
+//                    for (int i = 1; i < flatteningNames.length - 1; i++) {
+//                        jgen.writeFieldName(flatteningNames[i]);
+//                        jgen.writeStartObject();
+//                    }
+//
+//                    jgen.writeFieldName(flatteningNames[flatteningNames.length - 1]);
+//                    writeValue(jgen, writer, value);
+//
+//                    for (int i = 1; i < flatteningNames.length - 1; i++) {
+//                        jgen.writeEndObject();
+//                    }
+//
+//                    jgen.writeEndObject();
+//                } else {
+//                    jgen.writeFieldName(jsonPropertyName);
+//                    writeValue(jgen, writer, value);
+//                }
+//            }
+//
+//            jgen.writeEndObject();
+//
+//            return;
+//        }
```

Comment:
It might be better to create a github issue for this and add this code there rather than have this code commented out.

Reply:
Yes, I will remove all dead code in the class to have a more in-depth investigation on supporting handling serialization of JsonFlatten classes/fields ourselves.

Reply:
#21505